crates/subspace-archiving/tests/integration/piece_reconstruction.rs

```diff
@@ -49,90 +40,63 @@ fn segment_reconstruction_works() {
         }
     }
 
-    let reconstructor = PiecesReconstructor::new(RECORDED_HISTORY_SEGMENT_SIZE, kzg).unwrap();
+    let reconstructor = PiecesReconstructor::new(kzg).unwrap();
 
     let flat_pieces = reconstructor.reconstruct_segment(&maybe_pieces).unwrap();
 
-    let reconstructed_pieces = flat_pieces_to_regular(&flat_pieces);
-
-    assert_eq!(reconstructed_pieces.len(), PIECES_IN_SEGMENT as usize);
-    pieces.iter().zip(reconstructed_pieces.iter()).for_each(
-        |(original_piece, reconstructed_piece)| {
-            assert_eq!(
-                blake2b_256_254_hash(original_piece.as_ref()),
-                blake2b_256_254_hash(reconstructed_piece.as_ref())
-            );
-        },
-    );
+    assert_eq!(flat_pieces.len(), PIECES_IN_SEGMENT as usize);
+    archived_segments
+        .into_iter()
+        .next()
+        .unwrap()
+        .pieces
+        .iter()
+        .zip(flat_pieces.iter())
+        .for_each(|(original_piece, reconstructed_piece)| {
+            assert_eq!(original_piece, reconstructed_piece);
```

Comment:
It does, but we don't really expect it to happen unless we have a regression and while working on it I found it useful to print the bytes since I can see from the comparison what went wrong. At some point everything except witness was correct and it was helpful to see that only last 48 bytes are different, with hashes I just have a random garbage, with that goal you can just do `assert!(a == b)`, except clippy will likely suggest to replace with `assert_eq!()`.